src/linter/rules/seeded-randomness.ts

```diff
@@ -0,0 +1,140 @@
+		const metadata: SeededRandomnessMeta = {
+			consumerCalls:                 0,
+			callsWithFunctionProducers:    0,
+			callsWithAssignmentProducers:  0,
+			callsWithNonConstantProducers: 0
+		};
+		return {
+			results: elements.getElements()
+				// map and filter consumers
+				.flatMap(element => enrichmentContent(element, Enrichment.CallTargets).targets.map(target => {
+					metadata.consumerCalls++;
+					return {
+						range:         element.node.info.fullRange as SourceRange,
+						target:        target as Identifier,
+						searchElement: element
+					};
+				}))
+				// filter by calls that aren't preceded by a randomness producer
+				.filter(element => {
+					const producers = enrichmentContent(element.searchElement, Enrichment.LastCall).linkedIds
+						.map(e => dataflow.graph.getVertex(e.node.info.id) as DataflowGraphVertexFunctionCall);
+					const { assignment, func } = Object.groupBy(producers, f => assignmentSet.has(f.name) ? 'assignment' : 'func');
+					let nonConstant = false;
+
+					// function calls are already taken care of through the LastCall enrichment itself
+					for(const f of func ?? []) {
+						if(getValueOfArgument(dataflow.graph, f, { index: 0 }, [RType.Number, RType.String, RType.Logical]) !== undefined) {
+							metadata.callsWithFunctionProducers++;
+							return false;
+						} else {
+							nonConstant = true;
+						}
+					}
+
+					// assignments have to be queried for their destination
+					for(const a of assignment ?? []) {
+						const argIdx = assignmentArgIndexes.get(a.name) as number;
+						const dest = getReferenceOfArgument(a.args[argIdx]);
+						if(dest !== undefined && assignmentProducers.has(recoverName(dest, dataflow.graph.idMap) as string)){
+							if(getValueOfArgument(dataflow.graph, a, { index: 1-argIdx }, [RType.Number, RType.String, RType.Logical]) !== undefined) {
+								metadata.callsWithAssignmentProducers++;
+								return false;
+							} else {
+								nonConstant = true;
+							}
+						}
+					}
+
+					if(nonConstant) {
+						metadata.callsWithNonConstantProducers++;
+					}
+					return true;
+				})
+
+				.map(element => ({
+					certainty: LintingCertainty.Definitely,
+					function:  element.target,
+					range:     element.range
+				})),
+			'.meta': metadata
+		};
+	},
+	info: {
+		defaultConfig: {
+			randomnessProducers: [{ type: 'function', name: 'set.seed' }, { type: 'assignment', name: '.Random.seed' }],
+			randomnessConsumers: ['jitter', 'sample', 'sample.int', 'arima.sim', 'kmeans', 'princomp', 'rcauchy', 'rchisq', 'rexp', 'rgamma', 'rgeom', 'rlnorm', 'rlogis', 'rmultinom', 'rnbinom', 'rnorm', 'rpois', 'runif', 'pointLabel', 'some', 'rbernoulli', 'rdunif', 'generateSeedVectors'],
+		},
+		tags:        [LintingRuleTag.Robustness, LintingRuleTag.Reproducibility],
+		name:        'Seeded Randomness',
+		description: 'Checks whether randomness-based function calls are preceded by a random seed generation function like `set.seed`.'
```

Comment:
Probably, we should add how to... fix these errors? And separate cases in which there is _no_ producer vs. there is a _non-constant_ producer